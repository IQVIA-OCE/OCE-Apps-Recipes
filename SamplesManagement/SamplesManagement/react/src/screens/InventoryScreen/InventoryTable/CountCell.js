import React, { useContext } from 'react';
import { StyleSheet, View } from 'react-native';
import { TextInput } from '@oce-apps/apollo-react-native';
import { useField, useFormikContext } from 'formik';
import { INVENTORY_FORM_TYPE } from '../../../constants/Inventories';
import { InventoryContext } from '../InventoryContext';

const CountCell = ({ row }) => {
  const { values } = useFormikContext();
  const { editingType } = useContext(InventoryContext);
  const fieldIndex = values.products.findIndex(
    ({ lotNumberId }) => row.lotNumberId === lotNumberId
  );
  const [field, { touched, error }, helper] = useField(
    `products.${fieldIndex}`
  );

  return (
    <View style={styles.root} testID="CountCell">
      <TextInput
        style={styles.input}
        value={
          field &&
          field.value &&
          !isNaN(parseFloat(field.value.physicalQuantity)) &&
          field.value.physicalQuantity.toString() ? field.value.physicalQuantity : null
        }
        onChangeText={physicalQuantity =>
          helper.setValue({
            ...field.value,
            physicalQuantity: physicalQuantity ? physicalQuantity : null,
          })
        }
        error={
          touched && error && error.physicalQuantity
            ? error.physicalQuantity
            : null
        }
        readonly={
          editingType === INVENTORY_FORM_TYPE.preview ||
          editingType === INVENTORY_FORM_TYPE.editSaved
        }
        helperText={
          touched && error && error.physicalQuantity
            ? error.physicalQuantity
            : null
        }
      />
    </View>
  );
};
const styles = StyleSheet.create({
  root: {
    paddingVertical: 5,
  },
  input: {
    width: 80,
  },
});

export default CountCell;
