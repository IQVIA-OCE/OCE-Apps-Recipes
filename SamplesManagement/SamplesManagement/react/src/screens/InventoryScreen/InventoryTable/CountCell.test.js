import React from 'react';
import CountCell from './CountCell';
import { Formik } from 'formik';
import { TextInput } from 'apollo-react-native';
import { render, act } from '@testing-library/react-native';

const Component = () => (
  <Formik
    initialValues={{ id: '123', status: 'Saved', products: [{ id: '1', physicalQuantity: 2 }] }}
  >
    <CountCell row={{ id: '1' }} />
  </Formik>
);

describe('CountCell', () => {
  it('should render properly', async () => {
    const { container, getByTestId, rerender } = render(<Component />);

    act(() => {
      container.findByType(TextInput).props.onChangeText('1');
    });

    expect(getByTestId('CountCell')).toBeTruthy();

    rerender(<Component />)

    expect(getByTestId('CountCell')).toBeTruthy();
  });
});
