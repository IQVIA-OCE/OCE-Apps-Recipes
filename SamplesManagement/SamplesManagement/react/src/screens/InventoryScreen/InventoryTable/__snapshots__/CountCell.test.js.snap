// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountCell should render properly 1`] = `
<View
  style={
    Object {
      "paddingVertical": 5,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "width": 176,
        },
        Object {
          "width": 80,
        },
        Object {
          "opacity": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#D9D9D9",
            "borderRadius": 0,
            "borderWidth": 0,
          },
          Object {
            "width": 80,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "minHeight": 40,
            "overflow": "hidden",
            "paddingBottom": 0,
            "paddingTop": 0,
          }
        }
      >
        <TextInput
          editable={false}
          maxResizableWidth={750}
          minHeight={60}
          minWidth={0}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          paddingLeft={8}
          paddingRight={8}
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          selectionColor="#0768FD"
          style={
            Array [
              Object {
                "margin": 0,
                "textAlign": "left",
                "zIndex": 5,
              },
              undefined,
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "height": 40,
                "minHeight": 0,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingVertical": 5,
                "textAlignVertical": "center",
                "width": "100%",
              },
            ]
          }
          value="1"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`CountCell should render properly 2`] = `
<View
  style={
    Object {
      "paddingVertical": 5,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "width": 176,
        },
        Object {
          "width": 80,
        },
        Object {
          "opacity": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#D9D9D9",
            "borderRadius": 0,
            "borderWidth": 0,
          },
          Object {
            "width": 80,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "minHeight": 40,
            "overflow": "hidden",
            "paddingBottom": 0,
            "paddingTop": 0,
          }
        }
      >
        <TextInput
          editable={false}
          maxResizableWidth={750}
          minHeight={60}
          minWidth={0}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          paddingLeft={8}
          paddingRight={8}
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          selectionColor="#0768FD"
          style={
            Array [
              Object {
                "margin": 0,
                "textAlign": "left",
                "zIndex": 5,
              },
              undefined,
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "height": 40,
                "minHeight": 0,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingVertical": 5,
                "textAlignVertical": "center",
                "width": "100%",
              },
            ]
          }
          value="1"
        />
      </View>
    </View>
  </View>
</View>
`;
